### PR TITLE
Fixed include screwing up comment exclusion

### PR DESCRIPTION
**Pull request by thegrb93**
Fixes: #1176

### CODE REVIEW

**Comment by Plootie** [comment]
Tested with no problems and does fix #1176

**Comment by CaptainPRICE** [comment]
> Tested with no problems and does fix #1176

Not quiet... Don't just test it with how it is supposed to be used... Test it all way around.

I found there's a whole new bug with this:
**path/to/lib/#include.txt**
```golo
print("Hello!")
```

**main.txt**
```golo
#include"path/to/lib/#include" #[asd]# #dsa
```

Running `main.txt` prints `Hello!`.
Which is completely wrong, even if you look at the syntax highlighting fails:
image

Also, notice how file hierarchy (left pane) in E2 editor fails to read the filename of `#include.txt` (missing a hashtag), it is a different issue tho.

**Comment by Plootie** [comment]
> Not quiet... Don't just test it with how it is supposed to be used... Test it all way around.

i did...

I fail to see why 
> ```golo
> print("Hello!")
> ```
> 
> **main.txt**
> 
> ```golo
> #include"path/to/lib/#include" #[asd]# #dsa
> ```
Shouldn't print "Hello!"
Syntax highlighting only breaks when you dont have a space between include and the path but that should still be fixed

**Comment by CaptainPRICE** [comment]
> I fail to see why
> ...
> Shouldn't print "Hello!"

Because, it is not how it works right now (*master branch version*), according to the spec, it expects at least one space between `#include` and the path string. If there's no space, then an error is produced.
With this PR/pattern, it is basically now breaking that spec, as you can have no space between them.
You fail to see that minor change?

**Comment by thegrb93** [comment]
Include must be on a single line, no way around that.

File: lua/entities/gmod_wire_expression2/base/preprocessor.lua [hunk 1]
```diff
@@ -108,7 +108,7 @@ function PreProcessor:RemoveComments(line)
 					self.blockcomment = true -- We're now inside a block comment
 				elseif type == "normal" then -- We found a # instead
 					local pos = comments[i].pos
-					local incPos, incEndPos = line:sub(pos):find("#include%s*%b\"\"")
+					local incPos, incEndPos = line:sub(pos):find("#include%s+%b\"\"")
```
Comment (CaptainPRICE) [inline lua/entities/gmod_wire_expression2/base/preprocessor.lua]: Should have changed "%s+" to " +". A space, that is no other type of whitespace.

Comment (thegrb93) [reply lua/entities/gmod_wire_expression2/base/preprocessor.lua]: Tabs

Comment (CaptainPRICE) [reply lua/entities/gmod_wire_expression2/base/preprocessor.lua]: Wasn't previously allowed, you want to make this happen?
I'm cool with it. Just note what changes you're introducing.

Also, there should be an error when there's no path string on the same line then.

Comment (thegrb93) [reply lua/entities/gmod_wire_expression2/base/preprocessor.lua]: No

**Comment by CaptainPRICE** [comment]
> Include must be on a single line, no way around that.

Still able to have path string on the separate line without error:
```golo
#[ now we ]# #include   
#[ include]#"path/to/lib/#include"#[
#include ""#]#
```

Prints `Hello!`

-----

Here we go, on the roll, yet another bug catch:
```golo
#include  "path/to/lib/#include"print(_PI)"wtf"#wait wat
```

Same output as above (prints `Hello!`), but the thing is that `print(_PI)` is never run...

**Comment by CaptainPRICE** [comment]
(*cough* honestly, I don't think fixing this issue is this simple, if it were this simple with 4 lines of code, I would have done it myself.. *cough*)

**Comment by TomyLobo** [comment]
@CaptainPRICE Do you think it's worth the effort fixing it?
Are there maybe other, more important problems?
Keep in mind, we're all volunteers.

**Comment by thegrb93** [comment]
Something like this shouldn't be done sloppily. I appreciate the testing being done for me. This fix isn't actually much effort, it's just finicky and has to be perfect.

**Comment by thegrb93** [comment]
That said, directives were actually designed to consume the whole line so I think this PR and the issue its trying to fix is pretty stupid in the first place.

**Pull request closed without merging**